### ISSUE TITLE
Namespace and PURLs lost when importing SPDX BoM

### ISSUE THREAD

**Issue by MarkusTe** [DependencyTrack/dependency-track#1011]
### Current Behavior:
When importing an SPDX BoM which contains PURLS, the given PURLs and namespaces are lost.
Tried Tag and RDF, both transformed from YAML input via [https://tools.spdx.org/app/convert/](https://tools.spdx.org/app/convert/)

### Steps to Reproduce:

1. Create a new project
2. Import this sample BoM (.spdx)
```
SPDXVersion: SPDX-2.2
DataLicense: CC0-1.0
DocumentNamespace: http://spdx.org/spdxdocs/spdx-document-curl
DocumentName: curl-7.70.0
SPDXID: SPDXRef-DOCUMENT

## Creation Information
Creator: Organization: Example Inc.
Creator: Person: Thomas Steenbergen
Created: 2020-07-23T18:30:22Z
LicenseListVersion: 3.9
## Relationships
Relationship: SPDXRef-DOCUMENT DESCRIBES SPDXRef-Package-curl

## Package Information
PackageName: curl
SPDXID: SPDXRef-Package-curl
PackageVersion: 7.70.0
PackageDownloadLocation: git+https://github.com/curl/curl.git@53cdc2c963e33bc0cc1a51ad2df79396202e07f8
PackageHomePage: https://curl.haxx.se/
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: curl
PackageCopyrightText: <text>Copyright (c) 1996 - 2020, Daniel Stenberg, <daniel@haxx.se>, and many contributors, see the THANKS file.</text>
PackageDescription: <text>A command line tool and library for transferring data with URL syntax, supporting HTTP, HTTPS, FTP, FTPS, GOPHER, TFTP, SCP, SFTP, SMB, TELNET, DICT, LDAP, LDAPS, MQTT, FILE, IMAP, SMTP, POP3, RTSP and RTMP. libcurl offers a myriad of powerful features.</text>
ExternalRef: PACKAGE-MANAGER purl pkg:deb/debian/curl@7.70.0
FilesAnalyzed: false

## Package Information
PackageName: junit
SPDXID: SPDXRef-Package-junit
PackageVersion: 4.12
PackageDownloadLocation: https://mvnrepository.com/artifact/junit/junit
PackageLicenseConcluded: NOASSERTION
PackageLicenseDeclared: Apache-2.0
PackageCopyrightText: <text>Copyright</text>
PackageDescription: <text>A ..</text>
ExternalRef: PACKAGE-MANAGER purl pkg:maven/junit/junit@4.12
FilesAnalyzed: false
```
3. Update portfolio metrics
4. Check the component details --> no namespace, no PURL 
     --> no info about outdated component, no info about existing vulnerabilities
5. Manually add the namespace and PURL for junit
6. Update portfolio metrics
7. Project shows info about outdated component, info about existing vulnerabilities

### Expected Behavior:
When importig a BoM which has all coordinates (name, namespace, version), and a PURL, make sure DT updates the component details in the database.

### Environment:
  
  - Dependency-Track Version: 4.0.1
  - Distribution: Bundled Executable WAR
  - BOM Format & Version: SPDX 2.2
  - Database Server: DT standard database
  - Browser: Firefox 78.7.1esr, Chrome 89.0.4389.114

### Additional Details:
-

**Comment by MarkusTe** [comment]
Cross reference: [https://github.com/spdx/spdx-spec/issues/473](https://github.com/spdx/spdx-spec/issues/473).

**Comment by MarkusTe** [comment]
Is this possibly because DT uses an older SPDX Java library that doesn't yet support PURLS and CPEs?
https://github.com/DependencyTrack/dependency-track/issues/1007

**Comment by stevespringett** [comment]
Thanks for reporting. SPDX support is not very well tested as it's not widely used with Dependency-Track today. I have found and resolved the issue. The fix will be available in v4.3. There is likely many other SPDX related issues. Pull requests are highly encouraged.

**Comment by stevespringett** [comment]
@MarkusTe apologies, but due to #1053, this issue will not be resolved.
